stdlib/universe/map_test.go

```diff
@@ -123,6 +123,45 @@ func TestMap_Process(t *testing.T) {
 		want    []*executetest.Table
 		wantErr error
 	}{
+		{
+			name: `overwrite groupkey`,
+			spec: &universe.MapProcedureSpec{
+				Fn: interpreter.ResolvedFunction{
+					Scope: builtIns,
+					Fn:    executetest.FunctionExpression(t, `(r) => ({r with _stop: "a"})`),
+				},
+			},
+			data: []flux.Table{&executetest.Table{
+				KeyCols: []string{"_stop"},
+				ColMeta: []flux.ColMeta{
+					{Label: "_stop", Type: flux.TTime},
+					{Label: "_time", Type: flux.TTime},
+					{Label: "_value", Type: flux.TFloat},
+				},
+				Data: [][]interface{}{
+					{execute.Time(6), execute.Time(1), 1.0},
+					{execute.Time(6), execute.Time(2), 2.0},
+					{execute.Time(6), execute.Time(3), 3.0},
+					{execute.Time(6), execute.Time(4), 4.0},
+					{execute.Time(6), execute.Time(5), 5.0},
+				},
+			}},
+			want: []*executetest.Table{{
+				KeyCols: []string{"_stop"},
+				ColMeta: []flux.ColMeta{
+					{Label: "_stop", Type: flux.TString},
+					{Label: "_time", Type: flux.TTime},
+					{Label: "_value", Type: flux.TFloat},
+				},
+				Data: [][]interface{}{
+					{"a", execute.Time(1), 1.0},
+					{"a", execute.Time(2), 2.0},
+					{"a", execute.Time(3), 3.0},
+					{"a", execute.Time(4), 4.0},
+					{"a", execute.Time(5), 5.0},
+				},
+			}},
+		},
```

Comment:
Do we need a similar testcase for when `mergeKey` is true?

Reply:
Ah, yes I saw test cases for `mergeKey=true`, but I guess it's possible to have `mergeKey=true` and still overwrite it in the lambda. Good catch. I'll add one.

And in this particular case, because the group column is being overwritten, it will not be missing from `obj` and therefore the `if` branch from above will be taken.

Reply:
Ok test case for `mergeKey=true` added.